File: backend/tests/test_api.py
import requests
import sys
import json
from uuid import uuid4

BASE_URL = "http://localhost:8000/api/v1"

session = requests.Session()

def login():
    res = session.post(f"{BASE_URL}/auth/login", data={"username": "admin@stark.com", "password": "password123"})
    if res.status_code != 200:
        print("LOGIN FAILED", res.text)
        sys.exit(1)
    token = res.json()["access_token"]
    session.headers.update({"Authorization": f"Bearer {token}"})
    print("Login OK")

def test_cases():
    print("--- Testing Cases CRUD ---")
    res = session.get(f"{BASE_URL}/cases")
    if res.status_code != 200:
        print("List cases failed:", res.text)
        return
    
    new_case = {
        "title": "Suspicious Login Activity",
        "description": "Multiple failed login attempts from unknown IP",
        "severity": "HIGH",
        "priority": "HIGH",
        "status": "OPEN",
        "tags": ["brute_force", "identity"]
    }
    res = session.post(f"{BASE_URL}/cases", json=new_case)
    if res.status_code != 201:
        print("Create Case Failed:", res.status_code, res.text)
    else:
        case_id = res.json()["id"]
        print(f"Created Case: {case_id}")
        
        res = session.get(f"{BASE_URL}/cases/{case_id}")
        if res.status_code != 200: print("View Case Failed:", res.text)
        
        res = session.patch(f"{BASE_URL}/cases/{case_id}", json={"status": "INVESTIGATING", "severity": "CRITICAL"})
        if res.status_code != 200: print("Edit Case Failed:", res.text)
        
        res = session.delete(f"{BASE_URL}/cases/{case_id}")
        if res.status_code not in (200, 204, 403, 405, 404): print("Delete Case returned unexpected:", res.status_code, res.text)

def test_alerts():
    print("--- Testing Alerts ---")
    res = session.get(f"{BASE_URL}/alerts")
    if res.status_code != 200:
        print("List alerts failed:", res.text)
        return
    
    items = res.json()["items"]
    if items:
        alert_id = items[0]["id"]
        r = session.get(f"{BASE_URL}/alerts/{alert_id}")
        if r.status_code != 200: print("View alert failed:", r.text)
        r = session.patch(f"{BASE_URL}/alerts/{alert_id}", json={"status": "ACKNOWLEDGED"})
        if r.status_code != 200: print("Edit alert failed:", r.text)

def test_threat_intel():
    print("--- Testing Threat Intel ---")
    new_ioc = {
        "type": "IP_ADDRESS",
        "value": "198.51.100.12",
        "severity": "HIGH",
        "confidence": 85,
        "source": "Custom Upload"
    }
    res = session.post(f"{BASE_URL}/threat-intel/indicators", json=new_ioc)
    if res.status_code != 201:
        print("Create IOC Failed:", res.text)
    else:
        ioc_id = res.json()["id"]
        print("Created IOC:", ioc_id)
        r = session.patch(f"{BASE_URL}/threat-intel/indicators/{ioc_id}", json={"severity": "CRITICAL"})
        if r.status_code != 200: print("Edit IOC failed:", r.text)
        
        r = session.delete(f"{BASE_URL}/threat-intel/indicators/{ioc_id}")
        if r.status_code not in (200, 204, 405, 403): print("Delete IOC returned:", r.status_code, r.text)


if __name__ == "__main__":
    login()
    test_cases()
    test_alerts()
    test_threat_intel()
    print("Phase 1 Tests Completed.")

File: backend/tests/test_api.js
const BASE_URL = "http://localhost:8000/api/v1";

async function main() {
    // Login
    let res = await fetch(`${BASE_URL}/auth/login`, {
        method: "POST",
        body: new URLSearchParams({username: "admin@stark.com", password: "password123"})
    });
    if (!res.ok) { console.error("LOGIN FAILED", await res.text()); process.exit(1); }
    const token = (await res.json()).access_token;
    const h = { "Authorization": `Bearer ${token}`, "Content-Type": "application/json" };
    console.log("Login OK");

    // Cases CRUD
    console.log("--- Testing Cases CRUD ---");
    res = await fetch(`${BASE_URL}/cases`, {headers: h});
    if (!res.ok) console.error("List cases failed", await res.text());
    
    res = await fetch(`${BASE_URL}/cases`, {
        method: "POST", headers: h,
        body: JSON.stringify({
            title: "Suspicious Login Activity", description: "Multiple failed login attempts",
            severity: "HIGH", priority: 1, status: "OPEN", tags: ["brute_force"]
        })
    });
    if (!res.ok) { console.error("Create Case Failed", await res.text()); }
    else {
        const case_id = (await res.json()).id;
        console.log("Created Case:", case_id);
        
        res = await fetch(`${BASE_URL}/cases/${case_id}`, {headers: h});
        if (!res.ok) console.error("View Case Failed", await res.text());
        
        res = await fetch(`${BASE_URL}/cases/${case_id}`, {
            method: "PATCH", headers: h,
            body: JSON.stringify({status: "INVESTIGATING", severity: "CRITICAL"})
        });
        if (!res.ok) console.error("Edit Case Failed", await res.text());
        else console.log("Edit Case OK");
        
        res = await fetch(`${BASE_URL}/cases/${case_id}`, { method: "DELETE", headers: h });
        console.log("Delete Case Returned:", res.status);
    }

    // Alerts
    console.log("--- Testing Alerts ---");
    res = await fetch(`${BASE_URL}/alerts`, {headers: h});
    let items = (await res.json()).items;
    if (items && items.length > 0) {
        const alert_id = items[0].id;
        res = await fetch(`${BASE_URL}/alerts/${alert_id}`, {headers: h});
        if (!res.ok) console.error("View Alert Failed", await res.text());
        
        res = await fetch(`${BASE_URL}/alerts/${alert_id}`, {
            method: "PATCH", headers: h,
            body: JSON.stringify({status: "ACKNOWLEDGED"})
        });
        if (!res.ok) console.error("Edit Alert Failed", await res.text());
        else console.log("Edit Alert OK");
    }

    // Assets
    console.log("--- Testing Assets ---");
    res = await fetch(`${BASE_URL}/assets`, {headers: h});
    items = (await res.json()).items;
    if (items && items.length > 0) {
        const asset_id = items[0].id;
        res = await fetch(`${BASE_URL}/assets/${asset_id}`, {
            method: "PATCH", headers: h,
            body: JSON.stringify({criticality: "HIGH"})
        });
        if (!res.ok) console.error("Edit Asset Failed", await res.text());
        else console.log("Edit Asset OK");
    }

    // Threat Intel
    console.log("--- Testing Threat Intel ---");
    res = await fetch(`${BASE_URL}/threat-intel/indicators`, {
        method: "POST", headers: h,
        body: JSON.stringify({
            type: "IP", value: "198.51.100.12", severity: "HIGH", confidence: 85, source: "Custom"
        })
    });
    if (!res.ok) { console.error("Create IOC Failed", await res.text()); }
    else {
        const ioc_id = (await res.json()).id;
        console.log("Created IOC:", ioc_id);
        
        res = await fetch(`${BASE_URL}/threat-intel/indicators/${ioc_id}`, {
            method: "PATCH", headers: h,
            body: JSON.stringify({severity: "CRITICAL"})
        });
        if (!res.ok) console.error("Edit IOC Failed", await res.text());
        
        res = await fetch(`${BASE_URL}/threat-intel/indicators/${ioc_id}`, { method: "DELETE", headers: h });
        console.log("Delete IOC Returned:", res.status);
    }
}
main();
